src/WebJobs.Script/Utility.cs

```diff
@@ -57,7 +57,7 @@ public static class Utility
 #endif
 
         private static readonly Regex FunctionNameValidationRegex = new Regex(@"^[a-z][a-z0-9_\-]{0,127}$(?<!^host$)", RegexOptions.IgnoreCase | RegexOptions.CultureInvariant);
-        private static readonly Regex BindingNameValidationRegex = new Regex(string.Format("^([a-zA-Z][a-zA-Z0-9]{{0,127}}|{0})$", Regex.Escape(ScriptConstants.SystemReturnParameterBindingName)));
+        private static readonly Regex BindingNameValidationRegex = new Regex(string.Format("^(?!.*__)([a-zA-Z_][a-zA-Z0-9_]{{0,127}}|{0})$", Regex.Escape(ScriptConstants.SystemReturnParameterBindingName)));
```

Comment:
Curious what the `|{0}` part is giving us? Seems like `^(?!.*__)([a-zA-Z_][a-zA-Z0-9_]{0,127})$` is logically equivalent?

Reply:
The `|{0}` part was there previously, and it allows us to use the `$return` binding name. The check is "match the constraints or match argument 0", and in our case, argument 0 is the next parameter of `String.Format()`, which is `Regex.Escape(ScriptConstants.SystemReturnParameterBindingName)`. See [here](https://learn.microsoft.com/en-us/dotnet/api/system.string.format?view=net-9.0#system-string-format(system-string-system-object-system-object-system-object)).

Removing that part makes this test case fail: `[InlineData(ScriptConstants.SystemReturnParameterBindingName)]` (the constant resolves to `$return`).